commit bbf7b896a7eabb496567d80c3c1c9d6ee1fb26ef (HEAD -> master)
Author: Mike Gerdts <mike.gerdts@joyent.com>
Date:   2019-08-20T17:14:31+00:00 (7 weeks ago)
    
    TRITON-1873 TRITON-1692 broke linux-prepare-image for Alpine Linux
    Reviewed by: Pedro Palazón Candel <pedro@joyent.com>
    Approved by: Pedro Palazón Candel <pedro@joyent.com>

diff --git a/tools/prepare-image/linux-prepare-image b/tools/prepare-image/linux-prepare-image
index 6e355ad..b6ab08a 100755
--- a/tools/prepare-image/linux-prepare-image
+++ b/tools/prepare-image/linux-prepare-image
@@ -98,9 +98,14 @@ function cleanup_root() {
     history -c
     history -w || true
 
-    # Remove or set password for root.  If root-password-hash is not set, this
-    # is equivalent to "passwd -d root".
-    usermod -p "$(mdata_get_pi root-password-hash)" root
+    # Set or remove the root password. prepare-image:root-password-hash is only
+    # supported on those distros that support "usermod -p <hash> <user>".
+    local pw="$(mdata_get_pi root-password-hash)"
+    if [[ -n "$pw" ]]; then
+        usermod -p "$pw" root
+    else
+        passwd -d root
+    fi
 
     # Clean up history, ssh keys, etc.
     cleanup_homedir /root
